trio/_core/_unbounded_queue.py

```diff
@@ -128,10 +128,8 @@ def get_batch_nowait(self):
             await self._lot.park()
             return self._get_batch_protected()
         else:
-            try:
-                return self._get_batch_protected()
-            finally:
-                await _core.cancel_shielded_checkpoint()
+            await _core.cancel_shielded_checkpoint()
+            return self._get_batch_protected()
```

Comment:
I'm not sure this is correct – what if someone else steals the batch while we're rescheduling?

Of course this is `UnboundedQueue` so it doesn't really matter, we're deleting it ASAP anyway.

Reply:
Good point -- fixed.